.github/workflows/publish.yml

```diff
@@ -0,0 +1,29 @@
+name: Publish documentation
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v2
+    - name: Restore Coursier cache
+      uses: coursier/cache-action@v3
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Set up Node.js and Yarn
+      uses: actions/setup-node@v2-beta
+      with:
+        node-version: 12
+    - name: Publish docs
```

Comment:
@Baccata that's what I was referring to, I've had success in publishing the site using a separate action and the github token that Actions creates, i.e.

```yaml
      - name: Build site
        run: sbt "buildSite --destination $(pwd)/_site"
      - name: Publish gh-pages
        uses: peaceiris/actions-gh-pages@v3
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: ./_site
```

(ignore the actual sbt command, that's from my project)

We might do it this way to avoid having a deploy key in the secrets.